frontend/src/pages/Admin.js

```diff
@@ -3,25 +3,45 @@ import {useAuth} from "../auth/AuthProvider";
 import Main from "../components/Main";
 import Header from "../components/Header";
 import Page from "../components/Page";
-import {useState} from "react";
+import {useEffect, useState} from "react";
 import TextField from "../components/TextField";
 import Button from "../components/Button";
 import Error from "../components/Error";
 import Loading from "../components/Loading";
+import UserGallery from "../components/UserGallery";
+
 
 
 export default function Admin() {
-    const {user, getNewVideo} = useAuth()
+    const {user, getNewVideo, getAllUser, token} = useAuth()
     const [newVideoId, setNewVideoId] = useState("")
     const [loading, setLoading] = useState(false)
+    const [allUser, setAllUser] = useState([])
+
+    useEffect(() => {
+        getAllUser(token).then(setAllUser)
+            .catch(error => console.error(error))
+    }, [getAllUser, token])
+
+
+    const reloadUserPage = () =>{
+        getAllUser(token)
+            .then(setAllUser)
+    }
+
+    const eachUserList = allUser.map(fetchedUser => (
+        <UserGallery fetchedUserName = {fetchedUser.userName} key = {fetchedUser.id} reloadUserPage = {reloadUserPage}
+        />))
```

Comment:
mach das doch in eine eigene komponente.

Reply:
auch eine gute Idee

Reply:
gute Idee